api/src/main/java/com/capitalone/dashboard/auth/webhook/github/GithubWebHookRequestFilter.java

```diff
@@ -0,0 +1,132 @@
+        String githubEnterpriseHost = request.getHeader("X-GitHub-Enterprise-Host");
+
+        GitHubWebHookSettings gitHubWebHookSettings = parseAsGitHubWebHook(apiSettings.getGitHubWebHook());
+
+        String userAgentExpectedValue = null;
+        List<String> githubEnterpriseHostExpectedValues = new ArrayList<>();
+        if (gitHubWebHookSettings != null) {
+            userAgentExpectedValue = gitHubWebHookSettings.getUserAgent();
+            githubEnterpriseHostExpectedValues = gitHubWebHookSettings.getGithubEnterpriseHosts();
+        }
+
+        if (checkForEmptyValues(userAgent, githubEnterpriseHost, userAgentExpectedValue, githubEnterpriseHostExpectedValues)
+                || !userAgent.contains(userAgentExpectedValue)
+                || !checkGithubEnterpriseHost(githubEnterpriseHost, githubEnterpriseHostExpectedValues)) {
+            authenticated = false;
+            filterChain.doFilter(request, response);
+        } else {
+            authenticated = true;
+            super.doFilter(req, res, filterChain);
+        }
+    }
+
+    private boolean checkForEmptyValues(String userAgent, String githubEnterpriseHost, String userAgentExpectedValue,
+                                List<String> githubEnterpriseHostExpectedValues) {
+        if (HygieiaUtils.checkForEmptyStringValues(userAgent, githubEnterpriseHost, userAgentExpectedValue)
+                || CollectionUtils.isEmpty(githubEnterpriseHostExpectedValues)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    public Authentication attemptAuthentication(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) throws AuthenticationException {
+
+        Authentication authentication = githubWebHookAuthService.getAuthentication(httpServletRequest);
+
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+
+        return authentication;
+    }
+
+    @Override
+    protected void successfulAuthentication(HttpServletRequest request, HttpServletResponse response, FilterChain chain,
+                                            Authentication authResult) throws IOException, ServletException {
+        SecurityContextHolder.getContext().setAuthentication(authResult);
+        chain.doFilter(request, response);
+    }
+
+    @Override
+    protected void unsuccessfulAuthentication(HttpServletRequest request, HttpServletResponse response,
+                                              AuthenticationException failed) throws IOException, ServletException {
+
+        response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Github Webhook Authentication Failed");
+    }
+
+    protected boolean checkGithubEnterpriseHost(String githubEnterpriseHost, List<String> githubEnterpriseHostExpectedValues) {
+        String value = Optional.ofNullable(githubEnterpriseHostExpectedValues)
+                .orElseGet(Collections::emptyList).stream()
```

Comment:
This method could be as simple as githubEnterpriseHostExpectedValues.contains(value);